table/block_based/block_like_traits.h

```diff
@@ -32,14 +55,32 @@ class BlocklikeTraits<BlockContents> {
     return 0;
   }
 
-  static Cache::DeleterFn GetDeleter(BlockType block_type) {
+  static size_t SizeCallback(void* obj) {
+    assert(obj != nullptr);
+    BlockContents* ptr = reinterpret_cast<BlockContents*>(obj);
+    return ptr->data.size();
+  }
+
+  static Status SaveToCallback(void* obj, size_t offset, size_t size,
+                               void* out) {
+    assert(obj != nullptr);
+    BlockContents* ptr = reinterpret_cast<BlockContents*>(obj);
+    const char* buf = ptr->data.data();
+    assert(size == ptr->data.size());
```

Comment:
I also recommend renaming `size` to `length` here and in cache.h. (We are closer to "What length of rope would you like?" than "What size of rope would you like?")

It's also not clear whether `offset` refers to input or output. Perhaps `from_offset` and even `from_obj`.